### PR TITLE
Relax restriction that sandboxes cannot add columns (not sure if we need to merge)

### PR DESCRIPTION
**Pull request by camsaul**
Relax restriction that sandboxes cannot add columns -- instead, just rework the `:source-query` used for the sandbox to explicitly include `:fields` that limit it to only return columns present in the Table getting sandboxed (effectively ignoring the additional columns).

I'm not sure which sandboxing issues this closes specifically. I was initially optimistic and thought it fixed everything but it turns out Cypress was just not running tests. In combination with #14655 I think this should let us close out a few things (TBD)

### CODE REVIEW

File: enterprise/backend/src/metabase_enterprise/sandbox/models/group_table_access_policy.clj [hunk 1]
```diff
@@ -47,7 +47,7 @@
 
 (s/defn check-columns-match-table
   "Make sure the result metadata data columns for the Card associated with a GTAP match up with the columns in the Table
-  that's getting GTAPped. It's ok to remove columns, but you cannot add new columns. The base types of the Card
+  that's getting GTAPped. It's ok to remove columns, but added columns may be ignored. The base types of the Card
```
Comment (dpsutton) [inline enterprise/backend/src/metabase_enterprise/sandbox/models/group_table_access_policy.clj]: should this be `added columns will be ignored`?

File: enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 ;; TODO - not sure what other endpoints we might need, e.g. for fetching the list above but for a given group or Table
 
-#_(def ^:private AttributeRemappings
+#_(def ^:private Attributeemappings
```
Comment (dpsutton) [inline enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj:26]: what's the point of this change?

Comment (camsaul) [reply enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj:26]: Oops. I didn't mean to do that

File: enterprise/backend/src/metabase_enterprise/sandbox/models/group_table_access_policy.clj (hunk 1 above)
Comment (camsaul) [reply enterprise/backend/src/metabase_enterprise/sandbox/models/group_table_access_policy.clj]: Good catch

**Comment by dpsutton** [comment]
@camsaul in this PR you replace `:errors      {:card_id msg}` with `:message msg`. The `{:card_id msg}` was added in a PR that is still pending. Was this an intentional clobber of a pending change?

**Comment by camsaul** [comment]
@dpsutton yeah, I originally put `:errors {:card_id msg}` in there hoping that would cause the frontend to display the error but that ended up not being the case. So it ended up being unused. `:message` is actually what ends up getting used, some of our Ring middleware adds it when it catches exceptions... I didn't really need to add it here but I wanted to have it show up in the tests. I could have made these changes in the other PR but I didn't notice it at the time and only noticed it here when I had to go rework the tests after removing the restriction for adding columns

**Comment by camsaul** [comment]
Superseded by #14735

**Pull request closed without merging**